qiita_db/test/test_user.py

```diff
@@ -224,38 +219,85 @@ def test_set_info_bad_info(self):
             self.user.info = self.userinfo
 
     def test_default_analysis(self):
-        obs = self.user.default_analysis
-        self.assertEqual(obs, 4)
+        try:
+            qiita_config.portal = "QIITA"
+            obs = self.user.default_analysis
+            self.assertEqual(obs, 4)
+
+            qiita_config.portal = "EMP"
+            obs = self.user.default_analysis
+            self.assertEqual(obs, 8)
+        finally:
+            qiita_config.portal = self.portal
 
     def test_get_user_studies(self):
         user = User('test@foo.bar')
-        self.assertEqual(user.user_studies, {1})
+        try:
+            qiita_config.portal = "QIITA"
+            self.assertEqual(user.user_studies, {1})
+
+            qiita_config.portal = "EMP"
+            self.assertEqual(user.user_studies, set())
+        finally:
+            qiita_config.portal = self.portal
 
     def test_get_shared_studies(self):
         user = User('shared@foo.bar')
-        self.assertEqual(user.shared_studies, {1})
+        try:
+            qiita_config.portal = "QIITA"
+            self.assertEqual(user.shared_studies, {1})
+
+            qiita_config.portal = "EMP"
+            self.assertEqual(user.shared_studies, set())
+        finally:
+            qiita_config.portal = self.portal
 
     def test_get_private_analyses(self):
-        self.assertEqual(self.user.private_analyses, set([]))
+        user = User('test@foo.bar')
+        try:
+            qiita_config.portal = "QIITA"
+            self.assertEqual(user.private_analyses, set([1, 2]))
+
+            qiita_config.portal = "EMP"
+            self.assertEqual(user.private_analyses, set())
+        finally:
+            qiita_config.portal = self.portal
 
     def test_get_shared_analyses(self):
-        self.assertEqual(self.user.shared_analyses, set([]))
+        user = User('shared@foo.bar')
+        try:
```

Comment:
If the exception is deterministic, can it be caught explicitly? If it is stochastic, then can the test be designed to be deterministic? Same thing with the other try/except blocks